### PR TITLE
Enable/fix Windows testing, enable Mac/Windows in Github Actions: Closes #336

### LINKED ISSUE #336: testing fails on Windows
Running unit tests fails on Windows:

testing/test_pytest_html.py::TestHTML::test_utf8_longrepr[u'\x81'] FAILED             
and
testing/test_pytest_html.py::TestHTML::test_report_display_utf8 FAILED

I wonder if we can/should skip these tests on Windows or modify them in some way.

### PR DESCRIPTION
**Pull request by csm10495**
Fix the default encoding by patching open to always use utf-8. This is the default on *nix based systems, though is different on Windows (and possibly others). 

Also while here, add testing for Mac/Windows to Github Actions

### CODE REVIEW

**Comment by csm10495** [comment]
@ssbarnea can you check this?

**Review by ssbarnea — CHANGES_REQUESTED** [review]
You must use different names for jobs in order to avoid confusions, example: "py3x-(windows|macos|linux)".

**Comment by csm10495** [comment]
@ssbarnea I finally got around to fixing this. Check it again. Right now it fails because of https://github.com/actions/setup-python/issues/151 (I don't think its my change specifically). 

Also in order to merge I think you may need to change settings to change the required actions for merge (since I changed the names):
image

**Comment by gnikonorov** [comment]
Hey @csm10495 can you resolve merge conflicts, then I can take  a look

**Comment by csm10495** [comment]
Yep working on it and the commit rewrite.

File: pytest_html/extras.py [hunk 1]
```diff
@@ -10,10 +10,10 @@
 FORMAT_VIDEO = "video"
 
 
-def extra(content, format, name=None, mime_type=None, extension=None):
```
Comment (gnikonorov) [inline pytest_html/extras.py:13]: I think the rebase is a little off. This change already exists in master ( see https://github.com/pytest-dev/pytest-html/blob/master/pytest_html/extras.py#L13 ). It looks like this PR counts the merge of master as your changes

**Review by gnikonorov — CHANGES_REQUESTED** [review]

**Comment by csm10495** [comment]
Closing in favor of a new PR.

**Pull request closed without merging**